src/dsp_tools/commands/xmlupload/xmlupload.py

```diff
@@ -180,30 +200,45 @@ def _prepare_upload(
     return resources, permissions_lookup, stash
 
 
-def upload_resources(
-    upload_state: UploadState,
-    ingest_client: AssetClient,
-    project_client: ProjectClient,
-    list_client: ListClient,
-) -> None:
+def _upload_resources(clients: UploadClients, upload_state: UploadState) -> None:
     """
-    Actual upload of all resources to DSP.
+    Iterates through all resources and tries to upload them to DSP.
+    If a temporary exception occurs, the action is repeated until success,
+    and if a permanent exception occurs, the resource is skipped.
 
     Args:
+        clients: the clients needed for the upload (AssetClient, ProjectClient, ListClient)
         upload_state: the current state of the upload
-        ingest_client: ingest server client
-        project_client: a client for HTTP communication with the DSP-API
-        list_client: a client for HTTP communication with the DSP-API
+
+    Raises:
+        BaseException: in case of an unhandled exception during resource creation
+        XmlUploadInterruptedError: if the number of resources created is equal to the interrupt_after value
     """
     try:
```

Comment:
I would keep the try block shorter. `_upload_one_resource()` is the only thing that needs to be inside it

Reply:
No, at least to keep the meaning of the program unchanged, both the loop around the the `_upload_one_resource()` and the stash handling must be in there.  
But you're right, creating the resource client and these things doesn't need to be in the try.